### PR TITLE
Add TypeScript support

### PR DESCRIPTION
**Pull request by rossng**
**Describe what the PR does**    

This PR begins the process of adding TypeScript support in slate-transcript-editor.

**State whether the PR is ready for review or whether it needs extra work**    

Ready to review.

**Additional context**

I have tried to make a fairly minimal set of changes to get TypeScript working. Notably:

* Babel is replaced by the TypeScript compiler (.babelrc -> tsconfig.json)
* Storybook has been upgraded to v6, which has much better native support for TS
* The minimum number of types required to satisfy the compiler have been added
* For now, TypeScript is configured with `strict: false`, which greatly reduces the strictness of the compiler's checks
* I have tried to keep the structure of the output package very similar to what was there before (but with added type declaration files)

Please let me know if you have any questions or suggestions 🙂

### CODE REVIEW

File: src/components/2-Longer.stories.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { action } from '@storybook/addon-actions';
-import SlateTranscriptEditor from './index.js';
+import SlateTranscriptEditor from '.';
```
Comment (pietrop) [inline src/components/2-Longer.stories.tsx:3]: is this shorthand for import './index.js'?

Comment (rossng) [reply src/components/2-Longer.stories.tsx:3]: In short: basically yes.

Long answer:

In this case, `index.js` became `index.ts`. VSCode/the TS language server doesn't let you use the sytax `import x from 'file.ts'`, so there are two options here:

1. `import SlateTranscriptEditor from '.'`
2. `import SlateTranscriptEditor from './index'`

**Comment by pietrop** [comment]
Thanks for this PR, just wanted to give you a heads up, that have some other things at the moment that are taking priority, but will circle back to this ass soon as I get a chance.

**Comment by rossng** [comment]
Closing this as I have decided to attempt a more substantial refactor. You are, of course, free to use any of my contributions under a MIT licence 🙂

**Pull request closed without merging**